sys/net/application_layer/gcoap/forward_proxy.c

```diff
@@ -84,8 +89,34 @@ gcoap_listener_t forward_proxy_listener = {
     _request_matcher_forward_proxy
 };
 
+static void *_forward_proxy_thread_start(void *arg)
+{
+    (void)arg;
+
+    msg_t msg;
+
+    while (1) {
+        msg_receive(&msg);
+        client_ep_t *cep = (client_ep_t *)msg.content.ptr;
+        switch (msg.type) {
+            case FORWARD_PROXY_MSG_SEND:
+                gcoap_req_send((uint8_t *)cep->pdu.hdr, coap_get_total_len(&cep->pdu),
```

Comment:
> ~~When no response is received~~ and the client context has already been passed to the proxy thread it will never get deallocated. 

If this call right here fails for some reason the client context is not deallocated.

Reply:
for example if no memo can be allocated.